dotCMS/src/main/java/com/dotmarketing/filters/CMSUrlUtil.java

```diff
@@ -415,9 +415,18 @@ String xssCheck(String uri, String queryString) throws ServletException {
 	public String getURIFromRequest(HttpServletRequest request)
 			throws UnsupportedEncodingException {
 
-		return (request.getAttribute(CMS_FILTER_URI_OVERRIDE) != null) ? (String) request
+        String requestPath = request.getRequestURI();
+        try {
+            URI requestURI = new URI(requestPath);
+            requestPath = requestURI.getPath();
+        } catch (URISyntaxException e) {
+            Logger.error(this, "Couldn't get URL from request " + requestPath, e);
+        }
```

Comment:
I think all this logic should go to a private method then we can call that method here

https://github.com/dotCMS/core/pull/17785/files#diff-b01fee3e8841e6997654dbe68e5baf2dR429

this way if "request.getAttribute(CMS_FILTER_URI_OVERRIDE) != null)" that code is not ran

Reply:
Done